enhancements/oc/mirroring-manifests.md

```diff
@@ -0,0 +1,182 @@
+                ...
+                ├── 4.3.0-thanos -> mirror/v2/openshift/release/manifests/sha256:42ab3f59d3e769e88af89c30304b9cc77a76b66960747e2e1a6f8db097420858
+                ├── sha256:00ec58112784d340179e045bc70399ac1bd509ae43c4411e934ced23d016b2a1
+                ...
+                └── sha256:ffa87c992c5a2c51290e6d67cb609f29893dd147c31cff639ff6785ae7a1cfe2
+
+5 directories, 514 files
+```
+
+that can be applied to the local registry with the given `oc image mirror` command.
+
+With this enhancement, the output format would gain an additional directory `manifests` as a sibling to the current `v2` containing manifest files that can be applied to the local cluster with:
+
+```console
+$ oc apply -Rf mirror/manifests
+```
+
+The `manifests` directory may or may not contain additional subdirectories, so the recursive `-R` should be given for future-proofing.
+
+### Pushing to disk while pushing containers to the mirror registry
+
+Users mirroring a release before creating a cluster may want to [push both container images and manifests to disk](#pushing-to-disk).
+But they might also want to push container images directly to a mirror repository (as in [the *applying directly to the target cluster* case](#applying-directly-to-the-target-cluster) while still pushing manifests to disk (because there is not yet a cluster into which those manifests can be pushed).
+While you could address this use-case with [the *pushing to disk* flow](#pushing-to-disk) followed by an immediate `oc image mirror ...`, it is more convenient to have `oc adm release mirror ...` push to the registry directly.
+Because setting `--to-dir` adjusts the image source, this enhancement extends the command with a new `--manifests-to-dir` that can be used to set a manifest output directory without setting a container image output directory.
+When `--manifests-to-dir` is set, it takes precedence.
+When `--manifests-to-dir` is unset but `--to-dir` is set, the manifest directory is `${TO_DIR}/manifests`, as described in [the previous section](#pushing-to-disk).
+
+### User Stories
+
+#### Release-signature ConfigMaps
+
+FIXME
+
+#### In-cluster Cincinnati
+
+FIXME
+
+### Risks and Mitigations
+
+FIXME
+
+## Design Details
+
+### Test Plan
+
+FIXME
+
+### Graduation Criteria
+
+FIXME
+
+### Upgrade / Downgrade Strategy
+
+FIXME
+
+### Version Skew Strategy
+
+The proposal for [applying directly to the target cluster](#applying-directly-to-the-target-cluster) adds a new dependency on the cluster to which `oc` connects for Kubernetes activity.
+The explicit `--apply-manifests` argument requires users who have been using the previous `oc adm release mirror` implementation (which did not push manifests) to explicitly opt in to the new functionality, so there is no change that they accidentally pushing manifests into the wrong cluster.
+FIXME: check to see how this works if there are conflicts but neither `--force-conflicts` not `--overwrite` was set.
```

Comment:
I tested this with my implemented --apply-manifests option and the cluster just ignores the create and gives no error, warning, etc. I checked creationTimestamp and it does not "recreate" the ConfigMap either. 
We probably should have the --overwrite option and if not provided first check if the ConfigMap exists and if so give a warning that it will not be replaced without the --overwrite option.

Reply:
For reference, the way 'oc apply' current works WRT creation/update of a configmap is:
- If the given configmap, as defined by namespace and name, already exists, it will update it without the need for any additional option. The resulting message is different in that it says the configmap was "configured" vs "created".
- The configmap is truly updated as opposed to replaced. This has the following behavior WRT the binaryData field. If the name on binaryData field is changed, e.g. changed sha256-3a516480dfd68e0f87f702b4d7bdd6f6a0acfdac5cd2e9767b838ceede34d70d-1 to sha256-3a516480dfd68e0f87f702b4d7bdd6f6a0acfdac5cd2e9767b838ceede34d70d-4, it will create an additional binaryData entry for -4 leaving -1 as is. This is not the behavior we would want for 'oc adm release mirror --overwrite". We just want it to completely replace whatever is there with our new configmap.

Also, --overwrite would only apply when --apply-manifests option is specified. Currently 'oc adm release mirror' default behavior when writing images out to --to-dir is to simply overwrite what is currently there with no warning or indication that it is doing so. Make sense to have the newly added configmap output work in the same manner - simply overwrite whatever is there by default with no warning.